bundles/org.openhab.ui/web/src/components/sse-events-mixin.js

```diff
@@ -0,0 +1,127 @@
+import OhSheet from './widgets/modals/oh-sheet.vue'
+import OhPopover from './widgets/modals/oh-popover.vue'
+
+export default {
+  data () {
+    return {
+      eventSource: null,
+      audioContext: null
+    }
+  },
+  methods: {
+    startEventSource () {
+      const topicAudio = 'openhab/webaudio/playurl'
+      const commandItem = localStorage.getItem('openhab.ui:commandItem')
+      const topicCommand = `openhab/items/${commandItem || ''}/command`
+      let topics = null
+      if (localStorage.getItem('openhab.ui:webaudio.enable') === 'enabled') {
+        topics = topicAudio
+      }
+      if (commandItem) {
+        topics = topics ? `${topics},${topicCommand}` : topicCommand
+      }
+      if (!topics) return
+      this.eventSource = this.$oh.sse.connect(`/rest/events?topics=${topics}`, null, (event) => {
+        console.debug('Received SSE event: ' + JSON.stringify(event))
+        switch (event.topic) {
+          case topicAudio:
+            const topicParts = event.topic.split('/')
+            switch (topicParts[2]) {
+              case 'playurl':
+                this.playAudioUrl(JSON.parse(event.payload))
+                break
+            }
+            break
+          case topicCommand:
+            const payload = JSON.parse(event.payload)
+            const [command, ...segments] = payload.value.trim().split(':')
+            const arg = segments.join(':')
+            switch (command) {
+              case 'navigate':
+                this.$f7.views.main.router.navigate(arg)
+                break
+              case 'popup':
+              case 'popover':
+              case 'sheet':
+                if (arg.indexOf('page:') !== 0 && arg.indexOf('widget:') !== 0 && arg.indexOf('oh-') !== 0) {
+                  console.error('Action target is not of the format page:uid or widget:uid')
+                  return
+                }
+                console.debug(`Opening ${arg} in ${command} modal`)
+                let modalRoute = {
+                  url: arg + '/' + command,
+                  route: {
+                  }
+                }
+                if (command === 'popup') modalRoute.route.popup = { component: OhPopup }
+                if (command === 'popover') modalRoute.route.popup = { component: OhPopover }
+                if (command === 'sheet') modalRoute.route.popup = { component: OhSheet }
+                let modalProps = {
+                  props: {
+                    uid: arg,
+                    modalParams: {}
+                  }
+                }
+                this.$f7.views.main.router.navigate(modalRoute, modalProps)
+                break
+              case 'back':
+                this.$f7.views.main.router.back()
+                break
+              case 'reload':
+                window.location.reload()
+                break
+              case 'notification':
+                const toast = this.$f7.toast.create({
```

Comment:
I love your idea of having a command Item to control the UI!!

Just one very minor comment after having a look at the idea (not really at the code, I will leave the review to Yannick because he started it): IMO we should not use toasts for user-sent notifications here, I would consider them as only-used by the UI itself. I would rather use Framework7 notifications for this, see https://v5.framework7.io/docs/notification.

Reply:
Thanks @florian-h05 !  I did not think about f7 notifications, i agree, it would be a much better experience.